packages/renderer/client/render.ts

```diff
@@ -26,12 +28,28 @@ export const render = async (
       name: project.name,
     };
 
-    const frames = await renderer.getNumberOfFrames(settings);
+    const firstFrame = renderer.timeToFrame(startInSeconds);
+
+    let frames: number;
+    console.log('startInSeconds', startInSeconds);
+    console.log('endInSeconds', endInSeconds);
+    console.log('!isFinite', !isFinite(endInSeconds));
+    console.log('startInSeconds === 0', startInSeconds === 0);
+    if (startInSeconds === 0 && !isFinite(endInSeconds)) {
+      console.log('option 1');
+      frames = await renderer.getNumberOfFrames(settings);
+    } else {
+      console.log('option 2');
+      frames = renderer.timeToFrame(endInSeconds) - firstFrame + 1;
+    }
 
     // Get the range of frames to render (there can't be any overlap between workers)
     const framesPerWorker = Math.ceil(frames / totalNumOfWorkers);
-    const startFrame = framesPerWorker * workerId;
-    const endFrame = Math.min(startFrame + framesPerWorker, frames);
+    const startFrame = firstFrame + framesPerWorker * workerId;
```

Comment:
Not sure I understand this. Before your change, we calculate start and end based on the worker id. If we have 100 frames and 2 workers, each worker does 50 frames. I think it's not clear how this is overwriting this. Ideally, we only need to change the range of the video in one location (at the very beginning), and then we can still apply the "splitting" logic as we did before, just pretending that the overall video is shorter.